Sources/NIO/Socket.swift

```diff
@@ -152,21 +152,39 @@ typealias IOVector = iovec
             try Posix.writev(descriptor: $0, iovecs: iovecs)
         }
     }
-
+    
     /// Send data to a destination.
     ///
     /// - parameters:
     ///     - pointer: Pointer (and size) to the data to send.
+    ///     - controlBytes: Extra `cmsghdr` information.
     ///     - destinationPtr: The destination to which the data should be sent.
-    /// - returns: The `IOResult` which indicates how much data could be written and if the operation returned before all could be written (because the socket is in non-blocking mode).
+    ///     - destinationSize: The size of the destination address given.
+    /// - returns: The `IOResult` which indicates how much data could be written and if the operation returned before all could be written
+    /// (because the socket is in non-blocking mode).
     /// - throws: An `IOError` if the operation failed.
-    func sendto(pointer: UnsafeRawBufferPointer, destinationPtr: UnsafePointer<sockaddr>, destinationSize: socklen_t) throws -> IOResult<Int> {
+    func sendmsg(pointer: UnsafeRawBufferPointer,
+                 destinationPtr: UnsafePointer<sockaddr>,
+                 destinationSize: socklen_t,
+                 controlBytes: UnsafeMutableRawBufferPointer) throws -> IOResult<Int> {
+        var vec = iovec(iov_base: UnsafeMutableRawPointer(mutating: pointer.baseAddress!), iov_len: pointer.count)
+        
+        // Dubious const cast - it should be OK as there is no reason why this should get mutated
+        // just bad const declaration below us.
```

Comment:
This is really a pair of dubious casts: want to lift this comment up to cover both lines?